server/oauth.ts

```diff
@@ -0,0 +1,362 @@
+	| 'guardian.identity-api.newsletters.update.self'
+	| 'guardian.identity-api.user.read.self.secure'
+	| 'guardian.identity-api.user.update.self.secure'
+	| 'guardian.identity-api.user.username.create.self.secure'
+	| 'guardian.identity-api.consents.read.self'
+	| 'guardian.identity-api.consents.update.self'
+	| 'guardian.identity-api.cookies.create.self.secure';
+
+export const scopes: Scopes[] = [
+	'openid',
+	'profile',
+	'email',
+	'guardian.avatar-api.read.self',
+	'guardian.avatar-api.update.self',
+	'guardian.identity-api.newsletters.read.self',
+	'guardian.identity-api.newsletters.update.self',
+	'guardian.identity-api.user.read.self.secure',
+	'guardian.identity-api.user.update.self.secure',
+	'guardian.identity-api.user.username.create.self.secure',
+	'guardian.identity-api.consents.read.self',
+	'guardian.identity-api.consents.update.self',
+	'guardian.identity-api.cookies.create.self.secure',
+];
+
+export const ManageMyAccountOpenIdClient = async () => {
+	const oktaConfig = await getOktaConfig();
+	const issuer = joinUrl(
+		oktaConfig.orgUrl,
+		'/oauth2/',
+		oktaConfig.authServerId,
+	);
+	const OIDC_METADATA: IssuerMetadata = {
+		issuer,
+		authorization_endpoint: joinUrl(issuer, '/v1/authorize'),
+		token_endpoint: joinUrl(issuer, '/v1/token'),
+		jwks_uri: joinUrl(issuer, '/v1/keys'),
+		userinfo_endpoint: joinUrl(issuer, '/v1/userinfo'),
+		registration_endpoint: joinUrl(issuer, '/oauth2/v1/clients'),
+		introspection_endpoint: joinUrl(issuer, '/v1/introspect'),
+		revocation_endpoint: joinUrl(issuer, '/v1/revoke'),
+		end_session_endpoint: joinUrl(issuer, '/v1/logout'),
+	};
+
+	const OIDCIssuer = new Issuer(OIDC_METADATA);
+
+	return new OIDCIssuer.Client({
+		client_id: oktaConfig.clientId,
+		client_secret: oktaConfig.clientSecret,
+		redirect_uris: [`https://manage.${conf.DOMAIN}/oauth/callback`],
+	});
+};
+
+/**
+ * @param redirectUri - the redirect uri to use for the /authorize endpoint
+ * @param scopes (optional) - any scopes to use for the /authorize endpoint, defaults to ['openid']
+ * @param sessionToken (optional) - if provided, we'll use this to set the session cookie
+ * @param returnPath (optional) - if provided, we'll use this as the return path after the /authorize callback
+ */
+interface PerformAuthorizationCodeFlowOptions {
+	redirectUri: string;
+	scopes: Scopes[];
+	sessionToken?: string | null;
+	returnPath?: string;
+}
+
+/**
+ * @name performAuthorizationCodeFlow
+ * @description Helper method to perform the Authorization Code Flow
+ *
+ * Used for post authentication with the session token to set a session cookie.
+ *
+ * @param req - the express request object
+ * @param res - the express response object
+ * @param options - the options for the authorization code flow
+ * @returns 303 redirect to the okta /authorize endpoint
+ */
+export const performAuthorizationCodeFlow = async (
+	_req: Request,
+	res: Response,
+	{
+		sessionToken,
+		scopes = ['openid'],
```

Comment:
to avoid confusion I find it useful to give them a more specific name for variables holding scopes lists. For example instead of using a `scopes` variable we would name it `requiredScopes` or `requestedScopes` or `tokenScopes` and things like that.

Reply:
I think this is pretty clear from the docblock above the function, so I'll leave it as is.